docs/current_docs/user-guide/573039-cli-cookbook.mdx

```diff
@@ -0,0 +1,121 @@
+
+```shell
+dagger -m github.com/kpenfound/dagger-modules/golang@v0.1.5 call build --project https://github.com/dagger/dagger#main --args ./cmd/dagger file --path ./cmd/dagger export --path /tmp/dagger
+```
+
+## Start an interactive terminal for a container returned by a function
+
+To start an interactive terminal for a container returned by a function, use `dagger call FUNCTION-NAME terminal`. By default, the terminal is started with the `sh` shell, although this can be overridden by adding the `--cmd` argument.
+
+For example, to start an interactive terminal with the Wolfi base container returned by the `container` function of the `wolfi` module, use:
+
+```shell
+dagger call -m github.com/shykes/daggerverse/wolfi@v0.1.2 container --packages=cowsay terminal
+```
+
+To start the same terminal with the `zsh` shell, use:
+
+```shell
+dagger call -m github.com/shykes/daggerverse/wolfi@v0.1.2 container --packages=cowsay,zsh terminal --cmd zsh
+```
+
+## Start a service returned by a function
+
+To start a service returned by a function and have it forward traffic to a specified address via the host, use  `dagger call FUNCTION-NAME up`. If no port information is supplied, all of the service's ports are forwarded.
+
+By default, each port maps to the same port on the host. To specify a different mapping, use the additional `--ports` argument with a list of host/service port mappings. To bind ports randomly, use the `--random` argument.
+
+For example, to start a MariaDB service on host port 3306 (the default MariaDB port) with the `serve` function of the `mariadb` module, use:
+
+```shell
+dagger -m github.com/levlaz/daggerverse/mariadb@b56073942f5fa65c42a63a524fc9276f7f040f2f call serve up
+```
+
+To start the same service and map MariaDB port 3306 to host port 39000, use:
+
+```shell
+dagger -m github.com/levlaz/daggerverse/mariadb@b56073942f5fa65c42a63a524fc9276f7f040f2f call serve up --ports 39000:3306
+```
+
+To start the same service and map MariaDB port 3306 to a random port on the host, use:
+
+```shell
+dagger -m github.com/levlaz/daggerverse/mariadb@b56073942f5fa65c42a63a524fc9276f7f040f2f call serve up --random
+```
+
+## Pass a secret to a function as an argument
+
+To pass a secret to a function, source it from a host environment variable (`env:`), the host filesystem (`file:`), or a host command (`cmd:`).
+
+For example, pass a GitHub access token from an environment variable named `GITHUB_TOKEN` to the `run` function of the `gh` module:
+
+```shell
+dagger -m github.com/aweris/daggerverse/gh@99a1336f8091ff43bf833778a324de1cadcf25ac call run --token env:GITHUB_TOKEN --cmd "issue list --repo dagger/dagger"
+```
+
+Or, pass the token from a host file named `github-token.txt`:
+
+```shell
+dagger -m github.com/aweris/daggerverse/gh@99a1336f8091ff43bf833778a324de1cadcf25ac call run --token file:/home/admin/github-token.txt --cmd "issue list --repo dagger/dagger"
```

Comment:
@kpenfound neither this example or the next `cmd` one works, even though the `env` one works correctly. Error is 
```
Stderr:
Post "https://api.github.com/graphql": net/http: invalid header field value for "Authorization"
``` 
Any idea why? The file contains the token on a single line, no blank line after.

Reply:
Good catch, this looks like a bug with how secret files are parsed. Opening a fix